source/client-side-operations-timeout/tests/runCursorCommand.yml

```
@@ -0,0 +1,316 @@
+      - { _id: 5, x: 55 }
+
+tests:
+  - description: errors if timeoutMode is set without timeoutMS
+    operations:
+      - name: runCursorCommand
+        object: *db
+        arguments:
+          commandName: find
+          command: { find: *collection }
+          timeoutMode: cursorLifetime
+        expectError:
+          isClientError: true
+
+  - description: error if timeoutMode is cursorLifetime and cursorType is tailableAwait
+    operations:
+      - name: runCursorCommand
+        object: *db
+        arguments:
+          commandName: find
+          command: { find: *collection }
+          timeoutMode: cursorLifetime
+          cursorType: tailableAwait
+        expectError:
+          isClientError: true
+
+  - description: error if getMore maxTimeMS setting is greater than timeoutMS
+    operations:
+      - name: runCursorCommand
+        object: *db
+        arguments:
+          commandName: find
+          command: { find: *collection }
+          timeoutMS: 5
+          maxTimeMS: 10
+        expectError:
+          isClientError: true
+
+  - description: error if getMore maxTimeMS setting is equal to timeoutMS
+    operations:
+      - name: runCursorCommand
+        object: *db
+        arguments:
+          commandName: find
+          command: { find: *collection }
+          timeoutMS: 5
+          maxTimeMS: 5
+        expectError:
+          isClientError: true
+
+  # If timeoutMode is unset, it should default to CURSOR_LIFETIME and the time remaining after the find succeeds should be applied to the getMore
+  - description: Non-tailable cursor lifetime remaining timeoutMS applied to getMore if timeoutMode is unset
+    operations:
+      # Block find/getMore for 15ms.
+      - name: failPoint
+        object: testRunner
+        arguments:
+          client: *failPointClient
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 2 }
+            data:
+              failCommands: [find, getMore]
+              blockConnection: true
+              blockTimeMS: 15
+      # Run a find with timeoutMS=20 and batchSize=1 to force two batches, which will cause a find and a getMore to be sent.
+      # Both will block for 15ms so together they will go over the timeout.
+      - name: runCursorCommand
+        object: *db
+        arguments:
+          commandName: find
+          timeoutMS: 20
+          command: { find: *collection, batchSize: 2 }
+        expectError:
+          isTimeoutError: true
+    expectEvents:
+      - client: *client
+        events:
+          - commandStartedEvent:
+              commandName: find
+              command:
+                find: *collection
+                maxTimeMS: { $$type: [int, long] }
+          - commandStartedEvent:
+              commandName: getMore
+              command:
+                getMore: { $$type: [int, long] }
+                collection: *collection
+                maxTimeMS: { $$exists: false }
```

```suggestion
                maxTimeMS: { $$exists: true }
```

In Go, the context will be set with the value of `arguments.timeoutMS`. This timeout will propagate to the operation's write message. Since find_timeoutMS  >  failpoint_timeoutMS, I would expect the difference between these two to be applied to the successive operation (getMore).  In the given use case I would expect a maxTimeMS to be ~5 ms.

---

I agree, good catch, the test is looking for the timeoutMS to be applied to the getMore so this was just copy pasta error